relay-server/src/metric_stats.rs

```diff
@@ -61,27 +59,30 @@ impl MetricStats {
     }
 
     /// Tracks the metric volume and outcome for the bucket.
-    pub fn track_metric(&self, scoping: Scoping, bucket: Bucket, outcome: Outcome) {
+    pub fn track_metric<'a, V>(&self, scoping: Scoping, bucket: V, outcome: Outcome)
```

Comment:
Decided to switch the implementation to take any type that can be converted into a `BucketView` since we ended up thinking that the semantics of owning the supplied `Bucket` introduce complications in places where this code is called.